### ISSUE TITLE
crash on join world

### ISSUE THREAD

**Issue by TheNimbleNinja** [FabricMC/fabric#1499]
https://gist.github.com/TheNimbleNinja/dedc5342d92266e67cd03d756b7fc381

**Comment by warjort** [comment]
This hard to read error message says it can't save your world because you have a StructureFeature without a StructureConfig.

The only obvious modded behaviour in your stacktrace is this:
````
at me.shedaniel.someforgepatchesported.LenientUnboundedMapCodec.encode(LenientUnboundedMapCodec.java:69)
at me.shedaniel.someforgepatchesported.LenientUnboundedMapCodec.encode(LenientUnboundedMapCodec.java:18)
````
a port of a forge patch for when modded dimensions get removed.

It is more likely that you or one of your mods are not creating the structures correctly?
e.g. the mod is not providing a default config and one is not specified explicitly.

Some comments from yarn that might be relevant:
https://github.com/FabricMC/yarn/blob/1.17/mappings/net/minecraft/world/gen/chunk/StructuresConfig.mapping

The FabricStructureBuilder in the fabric api for creating these programmatically, requires the default config to be non-null.
https://github.com/FabricMC/fabric/blob/e369698318eb899183a821f9bb7c91c149c8ede3/fabric-structure-api-v1/src/main/java/net/fabricmc/fabric/api/structure/v1/FabricStructureBuilder.java#L163

So I would guess the feature is from a json file?

**Comment by warjort** [comment]
There is also this recent fix: #1495

**Comment by Technici4n** [comment]
If it happens again, please ask in `#player-support` on Discord.
